File: app/src/main/java/com/danilomendes/figurines/di/NetworkModule.java
package com.danilomendes.figurines.di;

import android.app.Application;

import com.danilomendes.figurines.network.NetworkManager;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

/**
 * Created by danilo on 14-10-2017.
 */
@Module
public class NetworkModule {

    private final String mBaseUrl;

    public NetworkModule(String baseUrl) {
        this.mBaseUrl = baseUrl;
    }

    @Provides
    @Singleton
    public NetworkManager provideNetworkManager(Application application) {
        return new NetworkManager(application, mBaseUrl);
    }
}

File: app/src/main/java/com/danilomendes/figurines/di/NetworkModule.kt
package com.danilomendes.figurines.di

import android.app.Application
import com.danilomendes.figurines.data.remote.NetworkManager
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

/**
 * Created by danilo on 14-10-2017.
 */
@Module
class NetworkModule(private val mBaseUrl: String) {

    @Provides
    @Singleton
    fun provideNetworkManager(application: Application): NetworkManager {
        return NetworkManager(application, mBaseUrl)
    }
}
